File: android/src/main/kotlin/cat/bcn/commonmodule/flutter/common_module_flutter/analytics/AnalyticsBridge.kt
package cat.bcn.commonmodule.flutter.common_module_flutter.analytics

import cat.bcn.commonmodule.analytics.AnalyticsWrapper
import io.flutter.plugin.common.EventChannel

class AnalyticsBridge : EventChannel.StreamHandler, AnalyticsWrapper {

    private var eventSink: EventChannel.EventSink? = null

    override fun logEvent(name: String, parameters: Map<String, String>) {
        eventSink?.success(mapOf("name" to name, "parameters" to parameters))
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        eventSink = events
    }

    override fun onCancel(arguments: Any?) {
        eventSink = null
    }
}

File: ios/Classes/analytics/AnalyticsBridge.swift
import OSAMCommon

class AnalyticsBridge: NSObject, FlutterStreamHandler, AnalyticsWrapper {
    private var eventSink: FlutterEventSink?
    
    func logEvent(name: String, parameters: [String : String]) {
        eventSink?(["name": name, "parameters": parameters])
    }
    
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        eventSink = nil
        return nil
    }
}
